godot-core/src/builtin/collections/packed_array.rs

```diff
@@ -414,29 +437,50 @@ macro_rules! impl_packed_array {
         #[doc = concat!("Creates a `", stringify!($PackedArray), "` from the given slice.")]
         impl From<&[$Element]> for $PackedArray {
             fn from(slice: &[$Element]) -> Self {
-                let mut array = Self::new();
-                let len = slice.len();
-                if len == 0 {
-                    return array;
+                if slice.is_empty() {
+                    return Self::new();
                 }
-                array.resize(len);
-                let ptr = array.ptr_mut(0);
-                for (i, element) in slice.iter().enumerate() {
-                    // SAFETY: The array contains exactly `len` elements, stored contiguously in memory.
-                    unsafe {
-                        // `GString` does not implement `Copy` so we have to call `.clone()`
-                        // here.
-                        *ptr.offset(to_isize(i)) = element.clone();
-                    }
+                let mut array = Self::default_with_size(slice.len());
+                // SAFETY: The array contains exactly `len` elements, stored contiguously in memory.
+                let dst = unsafe { std::slice::from_raw_parts_mut(array.ptr_mut(0), slice.len()) };
+                dst.clone_from_slice(slice);
+                array
+            }
+        }
+
+        #[doc = concat!("Creates a `", stringify!($PackedArray), "` from the given Rust array.")]
+        impl<const N: usize> From<[$Element; N]> for $PackedArray {
+            fn from(arr: [$Element; N]) -> Self {
+                if N == 0 {
+                    return Self::new();
+                }
+                let mut array = Self::default_with_size(N);
+
+                // SAFETY: The packed array contains exactly N elements and the source array will be forgotten.
+                unsafe {
+                    let arr = std::mem::ManuallyDrop::new(arr);
+                    array.move_from_slice(arr.as_ptr(), N);
```

Comment:
Sorry, I kind a missed that the parameter would also be `arr/array`.
Maybe you can name them `packed_array` and `array` to make it clearer.

Reply:
```rs
let arr = std::mem::ManuallyDrop::new(arr);
```
This is not unsafe and should thus be outside the block.

Reply:
Done